File: examples/dotNET/2. Functions/AVI/AVI Read Write With Data Example/vb/Form2.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports NationalInstruments.Vision.MeasurementStudioDemo

Partial Public Class Form2
    Inherits Form
    Private form1 As Form1
    Public Sub New(ByVal owner As Form1)
        InitializeComponent()
        form1 = owner
    End Sub

    Public ReadOnly Property Graph() As AviSampleDataGraph
        Get
            Return AviSampleDataGraph1
        End Get
    End Property

    Public ReadOnly Property FrameSlider() As GenericSlider
        Get
            Return GenericSlider1
        End Get
    End Property

    Public ReadOnly Property FrameNum() As NumericUpDown
        Get
            Return NumericUpDown1
        End Get
    End Property

    Public ReadOnly Property CurTime() As TextBox
        Get
            Return curTimeBox
        End Get
    End Property

    Private Sub NumericUpDown1_ValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles NumericUpDown1.ValueChanged
        If FrameSlider.Value <> CType(NumericUpDown1.Value, Double) Then
            FrameSlider.Value = NumericUpDown1.Value
        End If
    End Sub

    Private Sub GenericSlider1_ValueChanged(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles GenericSlider1.ValueChanged
        FrameNum.Value = CType(FrameSlider.Value, Decimal)
        form1.ReadFrame(CInt(FrameSlider.Value))
    End Sub

    Private Sub AviSampleDataGraph1_AfterCursorMove(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles AviSampleDataGraph1.AfterCursorMove
        If FrameSlider.Value <> Graph.CursorValue Then
            FrameSlider.Value = Graph.CursorValue
        End If
    End Sub
End Class

File: examples/dotNET/2. Functions/AVI/AVI Read Write With Data Example/cs/Backup/Form2.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using NationalInstruments.Vision.MeasurementStudioDemo;

namespace AVI_Read_Write_With_Data_Example
{
    public partial class Form2 : Form
    {
        private Form1 form1;
        public Form2(Form1 owner)
        {
            InitializeComponent();
            form1 = owner;
        }

        public AviSampleDataGraph Graph
        {
            get { return aviSampleDataGraph1; }
        }

        public GenericSlider FrameSlider
        {
            get { return genericSlider1; }
        }

        public NumericUpDown FrameNum
        {
            get { return numericUpDown1; }
        }

        public TextBox CurTime
        {
            get { return curTime; }
        }

        private void numericUpDown1_ValueChanged(object sender, EventArgs e)
        {
            if (FrameSlider.Value != (double)numericUpDown1.Value)
            {
                FrameSlider.Value = (double)numericUpDown1.Value;
            }
        }

        private void aviSampleDataGraph1_AfterCursorMove(object sender, EventArgs e)
        {
            if (FrameSlider.Value != Graph.CursorValue)
            {
                FrameSlider.Value = Graph.CursorValue;
            }
        }

        private void genericSlider1_ValueChanged(object sender, EventArgs e)
        {
            FrameNum.Value = (decimal)FrameSlider.Value;
            form1.ReadFrame((uint)FrameSlider.Value);
        }
    }
}